### PR TITLE
Update documentation for update_policy fields

### PR DESCRIPTION
**Pull request by JaylonmcShan03**
This pr addresses - https://github.com/hashicorp/terraform-provider-google/issues/15202

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement

```

### CODE REVIEW

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown [hunk 1]
```diff
@@ -221,19 +221,19 @@ update_policy {
...
 
 * `type` - (Required) - The type of update process. You can specify either `PROACTIVE` so that the instance group manager proactively executes actions in order to bring instances to their target versions or `OPPORTUNISTIC` so that no action is proactively executed but the update will be performed as part of other actions (for example, resizes or recreateInstances calls).
 
-* `max_surge_fixed` - (Optional), The maximum number of instances that can be created above the specified targetSize during the update process. Conflicts with `max_surge_percent`. If neither is set, defaults to 1
+* `max_surge_fixed` - (Optional), Specifies a fixed number of VM instances. This must be a positive integer.
```
Comment (BBBmau) [inline mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:230]: At first glance I'm not really sure where these documentation updates are coming from. Are the docs old and needing to be updated? What's the reference for the new documentation that you're adding?

Comment (JaylonmcShan03) [reply mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:230]: Here is the doc - https://cloud.google.com/compute/docs/reference/rest/v1/instanceGroupManagers#InstanceGroupManager.FIELDS.update_policy

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb [hunk 2]
```diff
@@ -293,31 +293,31 @@ func ResourceComputeInstanceGroupManager() *schema.Resource {
...
-							Description:   `The maximum number of instances that can be created above the specified targetSize during the update process. Conflicts with max_surge_percent. If neither is set, defaults to 1`,
+							Description:   `Specifies a fixed number of VM instances. This must be a positive integer.`,
 						},
 
 						"max_surge_percent": {
 							Type:          schema.TypeInt,
 							Optional:      true,
 							ConflictsWith: []string{"update_policy.0.max_surge_fixed"},
 							ValidateFunc:  validation.IntBetween(0, 100),
-							Description:   `The maximum number of instances(calculated as percentage) that can be created above the specified targetSize during the update process. Conflicts with max_surge_fixed.`,
```
Comment (BBBmau) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb:304]: We'll want to include `Conflicts with <-attribute->.`, this description would make sense to have if this resource had `max_surge_calculated` attribute. Which as of right now doesn't include it.

The same change will need to be done on `max_unavailable_percent` since it's missing `max_unavailable_calculated`

**Review by BBBmau — CHANGES_REQUESTED** [review]

**Review by BBBmau — CHANGES_REQUESTED** [review]
We'll want to include this from the docs: `At least one of either maxSurge or maxUnavailable must be greater than 0.`

https://github.com/GoogleCloudPlatform/magic-modules/pull/10935/files#diff-f01f6698125e0a7cd08298c6a6e56df40709cad4caf8086f14c12760652d0c78L262-L267

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb (hunk 2 above)
Comment (JaylonmcShan03) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb:304]: Added Conflicts with to the max_surge_percent attribute

**Comment by JaylonmcShan03** [comment]
> We'll want to include this from the docs: `At least one of either maxSurge or maxUnavailable must be greater than 0.`
> 
> https://github.com/GoogleCloudPlatform/magic-modules/pull/10935/files#diff-f01f6698125e0a7cd08298c6a6e56df40709cad4caf8086f14c12760652d0c78L262-L267

Will do
"Conflicts with" has been added to the requested attributes.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown [hunk 3]
```diff
@@ -221,19 +221,19 @@ update_policy {
 }
 ```
 
-* `minimal_action` - (Required) - Minimal action to be taken on an instance. You can specify either `NONE` to forbid any actions, `REFRESH` to update without stopping instances, `RESTART` to restart existing instances or `REPLACE` to delete and create new instances from the target template. If you specify a `REFRESH`, the Updater will attempt to perform that action only. However, if the Updater determines that the minimal action you specify is not enough to perform the update, it might perform a more disruptive action.
```
Comment (BBBmau) [inline mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:224]: We can revert this since the fix is more focused on both `max_surge_percent` and `max_surge_fixed` being wrong.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb [hunk 4]
```diff
@@ -293,31 +293,31 @@ func ResourceComputeInstanceGroupManager() *schema.Resource {
...
 						"max_unavailable_fixed": {
 							Type:          schema.TypeInt,
 							Optional:      true,
 							Computed:      true,
 							ConflictsWith: []string{"update_policy.0.max_unavailable_percent"},
-							Description:   `The maximum number of instances that can be unavailable during the update process. Conflicts with max_unavailable_percent. If neither is set, defaults to 1.`,
+							Description:   `Specifies a fixed number of VM instances. This must be a positive integer.`,
 						},
-
+						// do changes
```
Comment (BBBmau) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb:314]: remove comment

**Review by BBBmau — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb (hunk 4 above)
Comment (JaylonmcShan03) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb]: Removed "TODO CHANGES" comment, and fixed "minimal_action" description.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown (hunk 3 above)
Comment (JaylonmcShan03) [reply mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:224]: Fixed the description

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb [hunk 5]
```diff
@@ -293,31 +293,31 @@ func ResourceComputeInstanceGroupManager() *schema.Resource {
 							Optional:      true,
 							Computed:      true,
 							ConflictsWith: []string{"update_policy.0.max_surge_percent"},
-							Description:   `The maximum number of instances that can be created above the specified targetSize during the update process. Conflicts with max_surge_percent. If neither is set, defaults to 1`,
+							Description:   `Specifies a fixed number of VM instances. This must be a positive integer. Conflicts with max_surge_percent. If neither is set, defaults to 1`,
```
Comment (askubis) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager.go.erb:296]: If I understand correctly, the part that the issue reported was " If neither is set, defaults to 1", and I think this problem will remain.

**Review by askubis — COMMENTED** [review]
generally, when changing the IGM resource, please also change the regional counterpart - google_compute_region_instance_group_manager - it has the same update_policy structure with all the fields zonal IGM has.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown (hunk 3 above)
Comment (BBBmau) [inline mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:224]: revert this change since we want to keep the `REFRESH` instead of just REFRESH to provide a nicer view for users looking at the docs.

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown [hunk 6]
```diff
@@ -221,19 +221,19 @@ update_policy {
...
 
 * `most_disruptive_allowed_action` - (Optional) - Most disruptive action that is allowed to be taken on an instance. You can specify either NONE to forbid any actions, REFRESH to allow actions that do not need instance restart, RESTART to allow actions that can be applied without instance replacing or REPLACE to allow all possible actions. If the Updater determines that the minimal update action needed is more disruptive than most disruptive allowed action you specify it will not perform the update at all.
 
 * `type` - (Required) - The type of update process. You can specify either `PROACTIVE` so that the instance group manager proactively executes actions in order to bring instances to their target versions or `OPPORTUNISTIC` so that no action is proactively executed but the update will be performed as part of other actions (for example, resizes or recreateInstances calls).
 
-* `max_surge_fixed` - (Optional), The maximum number of instances that can be created above the specified targetSize during the update process. Conflicts with `max_surge_percent`. If neither is set, defaults to 1
```
Comment (BBBmau) [inline mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:230]: The same here within the description of max_surge_percent

File: mmv1/third_party/terraform/website/docs/r/compute_region_instance_group_manager.html.markdown [hunk 7]
```diff
@@ -236,13 +236,13 @@ update_policy {
 
 * `instance_redistribution_type` - (Optional) - The instance redistribution policy for regional managed instance groups. Valid values are: `"PROACTIVE"`, `"NONE"`. If `PROACTIVE` (default), the group attempts to maintain an even distribution of VM instances across zones in the region. If `NONE`, proactive redistribution is disabled.
 
-* `max_surge_fixed` - (Optional), The maximum number of instances that can be created above the specified targetSize during the update process. Conflicts with `max_surge_percent`. It has to be either 0 or at least equal to the number of zones.  If fixed values are used, at least one of `max_unavailable_fixed` or `max_surge_fixed` must be greater than 0.
```
Comment (BBBmau) [inline mmv1/third_party/terraform/website/docs/r/compute_region_instance_group_manager.html.markdown:239]: here as well.

**Review by BBBmau — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown (hunk 6 above)
Comment (JaylonmcShan03) [reply mmv1/third_party/terraform/website/docs/r/compute_instance_group_manager.html.markdown:230]: Fixed

File: mmv1/third_party/terraform/website/docs/r/compute_region_instance_group_manager.html.markdown (hunk 7 above)
Comment (JaylonmcShan03) [reply mmv1/third_party/terraform/website/docs/r/compute_region_instance_group_manager.html.markdown:239]: Fixed

**Review by BBBmau — APPROVED** [review]
Thanks for the contribution!
